### PR TITLE
create new topic to forward password management events

### PR DESCRIPTION
**Pull request by pansbro12**
Draft for new topic that will allow CBC to access the password management events produced by auth-customer.iam-credentials-v1

### CODE REVIEW

File: dev-aws/kafka-shared/iam/iam.tf [hunk 1]
```diff
@@ -233,3 +233,28 @@ resource "kafka_topic" "iam_revoked_v1" {
...
+    }
+}
+
+module "iam_password_change_processor" {
+    source  = "../../../mmodules/tls-app"
+    cert_comon_name= "auth-customer/password-change-processor"
+    produce_topics = [kafka_topic.iam_password_change_forwarder_v1.name]
+    consume_topics = [(kafka_topic.iam_credentials_v1)]
+    consume_groups = ["iam.password-change-forwarder"]
+}
```
Comment (Nesze) [inline dev-aws/kafka-shared/iam/iam.tf:260]: wdyt of the name `credentials-public-events-mapper` for the service? 🤔

File: dev-aws/kafka-shared/iam/iam.tf [hunk 2]
```diff
@@ -233,3 +233,28 @@ resource "kafka_topic" "iam_revoked_v1" {
     "cleanup.policy"    = "delete"
   }
 }
+
+
+resource "kafka_topic" "iam_password_change_forwarder_v1" {
+    name = "auth.iam-password-change-forwarder-v1"
```
Comment (Nesze) [inline dev-aws/kafka-shared/iam/iam.tf:239]: Service runs in `auth-customer` ns and think prefix should match that.

File: dev-aws/kafka-shared/iam/iam.tf [hunk 3]
```diff
@@ -233,3 +233,28 @@ resource "kafka_topic" "iam_revoked_v1" {
...
+        # retain 100MB on each partition
+        "retention.bytes" = "104857600"
+        # keep data for 7 days
+        "retention.ms" = "604800000"
+        # allow max 1 MB for a message
+        "max.message.bytes" = "1048576"
+        "compression.type"  = "zstd"
+        "cleanup.policy"    = "delete"
+    }
+}
```
Comment (Nesze) [inline dev-aws/kafka-shared/iam/iam.tf:252]: I think we can use the [internal topic](https://github.com/utilitywarehouse/kafka-cluster-config/pull/202/files#diff-de69f762692bb71ed5f21bb386f47daa1aa19018d80b3bdfe72866dc22773ec3R31-R45) internal topic as the blueprint fro the public one.

File: dev-aws/kafka-shared/iam/iam.tf (hunk 1 above)
Comment (pansbro12) [reply dev-aws/kafka-shared/iam/iam.tf:260]: Yeah thats much better, I wasnt a fan of password_change_processor

File: dev-aws/kafka-shared/iam/iam.tf [hunk 4]
```diff
@@ -233,3 +233,28 @@ resource "kafka_topic" "iam_revoked_v1" {
     "cleanup.policy"    = "delete"
   }
 }
+
+
+resource "kafka_topic" "iam_credentials_public_events_mapper_v1" {
+    name = "auth-customer.credentials-public-events-mapper-v1"
```
Comment (Nesze) [inline dev-aws/kafka-shared/iam/iam.tf:239]: Topic name for the private queue is `auth-customer.iam-credentials-v1`, so am thinking something like `auth-customer.iam-credentials-v1-public` or something similar would be more appropriate. Wdyt? 🤔

File: dev-aws/kafka-shared/iam/iam.tf [hunk 5]
```diff
@@ -233,3 +233,28 @@ resource "kafka_topic" "iam_revoked_v1" {
...
+        # keep data for 7 days
+        "retention.ms" = "604800000"
+        # allow max 1 MB for a message
+        "max.message.bytes" = "1048576"
+        "compression.type"  = "zstd"
+        "cleanup.policy"    = "delete"
+    }
+}
+
+module "iam_password_change_processor" {
```
Comment (Nesze) [inline dev-aws/kafka-shared/iam/iam.tf:254]: Maybe rename the module to the app name? e.g. `iam_credentials_public_event_mapper...`

**Review by Nesze — APPROVED** [review]

**Review by sbuliarca — APPROVED** [review]

File: dev-aws/kafka-shared/iam/iam.tf [hunk 6]
```diff
@@ -233,3 +233,28 @@ resource "kafka_topic" "iam_revoked_v1" {
     "cleanup.policy"    = "delete"
   }
 }
+
+
+resource "kafka_topic" "iam_credentials_v1_public" {
```
Comment (sbuliarca) [inline dev-aws/kafka-shared/iam/iam.tf:238]: I'm thinking if we can create this topic directly in the MSK cluster, but I see the app is producing to this one, but consuming from kafka_topic.iam_credentials_v1 which would be possible, but it needs different brokers in the deployment

**Pull request merged**